src/redux/actions/bridge.ts

```diff
@@ -151,8 +154,47 @@ export const updateBalances = ():
     }
 
     dispatch(updateGasBalances());
+    dispatch(updateFees());
   };
 
+// update fees for selected swap direction
+export const updateFees = ():
+  ThunkAction<Promise<void>, {}, {}, AnyAction> => async (
+    dispatch: ThunkDispatch<{}, {}, AnyAction>, getState,
+  ): Promise<void> => {
+    const {
+      bridge: {
+        swapDirection,
+      },
+      net: {
+        incentivizedOutboundChannelContract,
+        polkadotApi,
+      },
+    } = getState() as RootState;
+
+    const denomination = 10 ** 18;
+
+    try {
+      switch (swapDirection) {
+        case SwapDirection.EthereumToPolkadot:
+          // TODO: Proper handling of conversion
+          const erc20Dot: any = await incentivizedOutboundChannelContract!.methods.fee().call();
+          const erc20DotFormatted = (Number(erc20Dot) / denomination).toString();
```

Comment:
Problem here is that `erc20Dot` can be 256-bits (when converted from a string to BigNumber). However the `Number` type is only 64-bits, so there is a chance of truncation.

@musnit @wildmolasses @sumitsnk Any of you happen to know any ethereum-ecosystem JS libs for currency formatting?

Reply:
we should probably use ethers.utils.formatUnits > https://docs.ethers.io/v5/api/utils/display-logic/#unit-conversion

Reply:
And yeh should we be using Number or Balance from polkadotjs? irrespective, we do need a way to convert between BigNumber which we should use for ethereum assets and Number/Balance for polkadot assets with an explicit decision on how to handle truncation (eg: accept it and show it to the user, disallow it/block big numbers and show an error, etc)

Reply:
I have used the number and formatting libraries native to each chain.